### PR TITLE
feat: support clean old lifecycle before new update

### PR DESCRIPTION
**Pull request by ColdsteelRail**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by ColdsteelRail** [comment]
https://github.com/KusionStack/kuperator/issues/302

File: pkg/controllers/podopslifecycle/utils.go [hunk 1]
```diff
@@ -76,7 +77,8 @@ func IDToLabelsMap(pod *corev1.Pod) (map[string]map[string]string, map[string]in
 }
 
 // IsLifecycleOnPod returns true if the lifecycle with lifecycleId exist on pod, otherwise returns false
-func IsLifecycleOnPod(lifecycleId string, pod *corev1.Pod) (bool, error) {
+func IsLifecycleOnPod(lifecycleId string, obj client.Object) (bool, error) {
```
Comment (shaofan-hs) [inline pkg/controllers/podopslifecycle/utils.go:80]: Use `pod *corev1.Pod` is more suitable?

File: pkg/controllers/utils/podopslifecycle/utils.go [hunk 2]
```diff
@@ -85,6 +87,16 @@ func Begin(c client.Client, adapter LifecycleAdapter, obj client.Object, updateF
 	return false, nil
 }
 
+// BeginWithCleaningOld is used for an CRD Operator to begin a lifecycle with cleaning the old lifecycle
+func BeginWithCleaningOld(c client.Client, adapter LifecycleAdapter, obj client.Object, updateFunc ...UpdateFunc) (updated bool, err error) {
+	if podInUpdateLifecycle, err := podopslifecycleutil.IsLifecycleOnPod(adapter.GetID(), obj); err != nil {
+		return false, fmt.Errorf("fail to check %s PodOpsLifecycle on Pod %s/%s: %s", adapter.GetID(), obj.GetNamespace(), obj.GetName(), err)
+	} else if podInUpdateLifecycle {
+		return false, Undo(c, adapter, obj)
```
Comment (shaofan-hs) [inline pkg/controllers/utils/podopslifecycle/utils.go:95]: Undo, then Begin, and then return?

File: pkg/controllers/podopslifecycle/utils.go (hunk 1 above)
Comment (ColdsteelRail) [reply pkg/controllers/podopslifecycle/utils.go]: [IsLifecycleOnPod is called by ](https://github.com/KusionStack/kuperator/pull/295/commits/947ca3c8b1c14af6f76c5052a132f59718c472a2)

File: pkg/controllers/utils/podopslifecycle/utils.go (hunk 2 above)
Comment (ColdsteelRail) [reply pkg/controllers/utils/podopslifecycle/utils.go]: https://github.com/KusionStack/kuperator/pull/295/commits/947ca3c8b1c14af6f76c5052a132f59718c472a2

**Review by shaofan-hs — APPROVED** [review]

**Pull request merged**